scripts/openstack-update.sh

```diff
@@ -0,0 +1,108 @@
+
+if [ -z "$OPERATOR_NAMESPACE" ]; then
+    echo "Please set OPERATOR_NAMESPACE"; exit 1
+fi
+if [ -z "$NAMESPACE" ]; then
+    echo "Please set NAMESPACE"; exit 1
+fi
+
+OPENSTACK_OPERATOR_CSV=$(oc get csv -n $OPERATOR_NAMESPACE -o name | grep openstack-operator)
+OPENSTACK_VERSION_CR=$(oc get openstackversion -n $NAMESPACE -o name)
+
+if [ ${FAKE_UPDATE} = "true" ]; then
+    oc get $OPENSTACK_OPERATOR_CSV -o yaml -n $OPERATOR_NAMESPACE  > $OUTFILE
+    sed -i $OUTFILE -e "s|value: .*/$CONTAINERS_NAMESPACE/\(.*\)@.*|value: quay.io/$CONTAINERS_NAMESPACE/\1:$CONTAINERS_TARGET_TAG|g"
+    OPENSTACK_DEPLOYED_VERSION=$(oc get -n $NAMESPACE $OPENSTACK_VERSION_CR --template={{.spec.targetVersion}})
+    sed -i $OUTFILE -e "s|value: $OPENSTACK_DEPLOYED_VERSION|value: $OPENSTACK_VERSION|"
+
+    oc apply -f $OUTFILE
+fi
+
+oc project $NAMESPACE
+# wait until openstackVersion cr completes reconcile, status.availableVersion should be the same as VERSION
+oc wait $OPENSTACK_VERSION_CR --for=jsonpath='{.status.availableVersion}'=$OPENSTACK_VERSION --timeout=$TIMEOUT
+
+
+OPENSTACK_DEPLOYED_VERSION=$(oc get $OPENSTACK_VERSION_CR --template={{.spec.targetVersion}})
+
+cat <<EOF >openstackversionpatch.yaml
+    "spec": {
+      "targetVersion": "$OPENSTACK_VERSION"
+      }
+EOF
+
+oc patch $OPENSTACK_VERSION_CR  --type=merge  --patch-file openstackversionpatch.yaml
+
+# wait for ovn update on control plane
+oc wait $OPENSTACK_VERSION_CR --for=condition=MinorUpdateOVNControlplane --timeout=$TIMEOUT
+
+# start ovn update on data plane
+DATAPLANE_NODESET=$(oc get openstackdataplanenodeset -o name | awk -F'/' '{print $2}')
+DATAPLANE_DEPLOYMENT=$(oc get openstackdataplanedeployment -o name | awk -F'/' '{print $2}')
+
+cat <<EOF >edpm-deployment-ovn-update.yaml
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneDeployment
+metadata:
+  name: $DATAPLANE_DEPLOYMENT-ovn-update
+spec:
+  nodeSets:
+    - $DATAPLANE_NODESET
+  servicesOverride:
+    - ovn
+EOF
+
+oc create -f edpm-deployment-ovn-update.yaml
+
+oc get openstackdataplanedeployment
+# wait for ovn dataplane completes
+oc wait $OPENSTACK_VERSION_CR  --for=condition=MinorUpdateOVNDataplane --timeout=$TIMEOUT
+echo "MinorUpdateOVNDataplane completed"
+
+# wait for control plane update to complete
+oc wait $OPENSTACK_VERSION_CR --for=condition=MinorUpdateControlplane --timeout=$TIMEOUT
+echo "MinorUpdateControlplane completed"
+
+# start data plane plane update for rest of edpm services
+cat <<EOF >edpm-deployment-update.yaml
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneDeployment
+metadata:
+  name: $DATAPLANE_DEPLOYMENT-update
+spec:
+  nodeSets:
+    - $DATAPLANE_NODESET
+  servicesOverride:
+    - update
+EOF
+
+oc create -f edpm-deployment-update.yaml
+
+# wait for completion of minor update
+oc wait $OPENSTACK_VERSION_CR --for=condition=MinorUpdateDataplane --timeout=$TIMEOUT
+echo "MinorUpdate completed"
+
+oc get openstackdataplanedeployment -o yaml
```

Comment:
you might want to remove this line going forward, as it could produce a lot of output, depending on the number of deployments that have been done.

Reply:
Yeah, it may make sense to just echo 'You can see the results by ...' to provide help for how to do it, assuming that's a useful thing to do.

Reply:
I can limit to just one deployement, the last one created for edpm update. Wdyt?

Reply:
done.